### DISCUSSION TITLE
Reddit?

### DISCUSSION THREAD

**Discussion by joonid** [aclap-dev/video-downloadhelper#1011] [Help and Q&A]
### Your question

https://www.reddit.com/r/cats/comments/1e2hjzu/bruno_a_wonderful_transformation/
Windows 10
Firefox 128.0 (64-bit)

No media to process in the current tab
Click play on video to help detect files…

### Addon & CoApp version and details

Version 9.0.2.12 (stable), target mozilla, locale en-US.
Companion App installed (v2.0.19) Recheck
C:\Program Files (x86)\DownloadHelper CoApp\vdhcoapp.exe

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

With Windows 11 Pro 64-bit version 23H2 build 22631.3880, Firefox 128.0 64-bit, licensed VDH 9.0.2.14a1 beta, CoApp 2.0.19, VDH downloaded your entirely adorable cat video just fine.  And I didn't even have to click play on the video on the web page.

Maybe you should try the beta.

**Reply by GrampaWildWilly** [reply]
I did notice something kind of fishy about this:

![#01](https://github.com/user-attachments/assets/3f8252f9-bfa8-40d0-af66-c17167419aed)

Every variant appeared to be duplicated.  So I fished the HLS master manifest out of the Network Monitor & ran ffprobe on it:

[ffprobe.txt](https://github.com/user-attachments/files/16242335/ffprobe.txt)

That was a bit confusing to understand so I extracted the main information from the ffprobe report, & I rearranged & edited it a bit:

[ffprobe Edited.txt](https://github.com/user-attachments/files/16242343/ffprobe.Edited.txt)

Well how about that . . .  This video is indeed offered as 12 distinct variants, but there's only 6 distinct resolutions.  At each resolution, 2 variants are offered.  They differ in their bit rates, which is to say, one is of a higher quality than the other in each pair.  It looks to me like VDH is successfully unwinding this & presenting the variants in descending order of resolution, & within each resolution, in descending order of quality.

Also, there are only 2 audio tracks.  Stream 0:0 is offered with the variant at the lower bit rate & Stream 0:1 is offered with the higher bit rate.  In keeping with that, Stream 0:0 is only 66 kb/s while Stream 0:1 is 100 kb/s.  Neither one is particularly great but that's the way this is being presented.

I downloaded the first thing VDH offered.  That appears to be Program 11, consisting of video Stream 0:13 & audio Stream 0:1.

![#02](https://github.com/user-attachments/assets/6a6a9408-f594-405d-ac70-475754f6a851)

The audio bit rate of 129kbps is, I believe, a quirk of a default value in ffmpeg.  Since ffmpeg is the download engine inside VDH, that comment is applicable.  I believe ffmpeg uses a default audio bit rate of 128kbps.  (129kbps is close enough.  I find Windows & ffprobe/ffmpeg rarely agree exactly on bit rates.)  So the audio is bloated up a bit.  The original is only 100kbps so that's all there really is in this video.  These bit rates are variable anyway so the numbers are only a rough guideline at best.  Besides, I would defy anyone to say they could hear differences between 66kbps, 100kbps, & 128/129 kbps audio, especially with this particular video, in which the audio is mostly elevator music, a few cat vocalizations, & the usual doting gurglings a mother makes over a baby.

**Reply by GrampaWildWilly** [reply]
Here's something else I noticed.  VDH is claiming the highest resolution is 1440p.  It's not.  It's 1920p.  At least, the resolution is 1080x1920 & the p value is usually the vertical dimension.  The values VDH is providing are pretty much all made up, as if it has an internal list of p values & it's using those, instead of getting the values from the actual resolutions.  A bug for @paulrouget Paul to consider?  I wouldn't call it a very important bug.  But it does look like a bug.

**Reply by joonid** [reply]
GWWilly, thank you for the work you put in on this.  I completely related to your comment "That was a bit confusing to understand..."  I don't even know how to generate the Reddit screen capture that you posted. It's unfortunate that a VDH user needs to be a programmer/developer/engineer to make sense of the extension.  I don't care which screen resolution I download, or which of the two audio tracks; I just want the darn thing to work (without having to install Windows 11 on my system).

Please see my previous post - night be relevant here...
https://github.com/aclap-dev/video-downloadhelper/discussions/794

Regards
